src/scenes/ObservationsRouter/common/MonitoringPlotEditPhotos.tsx

```diff
@@ -0,0 +1,242 @@
+        data: { file, isOriginal: false, type: 'Quadrat', mediaKind: isVideoFile(file) ? 'Video' : 'Photo' },
+        type: 'new' as const,
+      };
+    });
+    setMediaItems((prevPhotos) => [...prevPhotos, ...newPhotos]);
+  }, []);
+
+  const getUpdatePhotoCaption = useCallback(
+    (index: number) => (caption: string) => {
+      const updatedPhotos = mediaItems.map((mediaItem, i) => {
+        if (index !== i) {
+          return mediaItem;
+        }
+
+        if (mediaItem.type === 'new') {
+          return {
+            ...mediaItem,
+            data: { ...mediaItem.data, caption },
+          };
+        } else {
+          return {
+            ...mediaItem,
+            data: { ...mediaItem.data, caption },
+            isModified: true,
+          };
+        }
+      });
+      setMediaItems(updatedPhotos);
+    },
+    [mediaItems]
+  );
+
+  const getDeletePhoto = useCallback(
+    (index: number) => () => {
+      const mediaItem = mediaItems[index];
+      if (mediaItem.type === 'new') {
+        // remove new photos completely
+        const updatedPhotos = mediaItems.filter((_, i) => index !== i);
+        setMediaItems(updatedPhotos);
+      } else {
+        // mark existing photos as deleted
+        const updatedPhotos = mediaItems.map((item, i) => {
+          if (index === i) {
+            return {
+              ...item,
+              isDeleted: true,
+            };
+          }
+          return item;
+        });
+        setMediaItems(updatedPhotos);
+      }
+    },
+    [mediaItems]
+  );
+
+  useEffect(() => {
+    if (observationResults) {
+      observationResults.plantingZones.forEach((zone) =>
+        zone.plantingSubzones.forEach((subzone) =>
+          subzone.monitoringPlots.forEach((plot) => {
+            if (plot.monitoringPlotId === monitoringPlotId) {
+              setMonitoringPlotResult(plot);
+              setMediaItems(plot.media.map((mediaElement) => ({ type: 'existing', data: mediaElement })));
+              return;
+            }
+          })
+        )
+      );
+    }
+  }, [observationResults, monitoringPlotId]);
+
+  const goToPhotosTab = useCallback(() => {
+    navigate(APP_PATHS.OBSERVATIONS);
+  }, [navigate]);
+
+  const savePhotos = useCallback(() => {
+    void (async () => {
+      try {
+        const promises = mediaItems.map((mediaItem) => {
+          if (mediaItem.type === 'new') {
+            const formData = new FormData();
+            formData.append('file', mediaItem.data.file);
+
+            const payloadData = {
+              caption: mediaItem.data.caption,
+              type: undefined,
+            };
+            formData.append('payload', new Blob([JSON.stringify(payloadData)], { type: 'application/json' }));
+
+            const payload: UploadOtherPlotMediaApiArg = {
+              observationId,
+              plotId: monitoringPlotId,
+              body: formData as any,
+            };
+            return upload(payload).unwrap();
```

Comment:
I had to add this unwrap to execute all the promises together in line 176 @tommylau523